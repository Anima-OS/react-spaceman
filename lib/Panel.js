"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const styles_1 = require("./styles");
exports.squishClass = {
    xxs: 'xxs',
    xs: 'xs',
    s: 'xs',
    m: 's',
    l: 'm',
    xl: 'l',
    xxl: 'xl',
};
// margin-bottom
const belowClasses = styles_1.addClassesForAllSizes('b', (name, value) => `${name} { margin-bottom: ${value}px }`);
// padding-left, padding-right
const xClasses = styles_1.addClassesForAllSizes('x', (name, value) => `${name} { padding-left: ${value}px; padding-right: ${value}px }`);
// padding-top, padding-bottom
const yClasses = styles_1.addClassesForAllSizes('y', (name, value) => `${name} { padding-top: ${value}px; padding-bottom: ${value}px }`);
// Inline
const inlineContainerClass = styles_1.addClass('i', name => `${name} { display: flex; flex-wrap: wrap }`);
const inlineClasses = styles_1.addClassesForAllSizes('i', (name, value) => `${name} > *:not(:last-child) { margin-right: ${value}px }`);
// Stack
const stackClasses = styles_1.addClassesForAllSizes('s', (name, value) => `${name} > *:not(:last-child) { margin-bottom: ${value}px }`);
styles_1.attachAllStyles();
function Panel({ below, inset, x, y, between, inline, squish, tag, tagProps, className, children }) {
    const xSize = inset || x;
    const ySize = inset || y;
    const classes = [
        className,
        below && belowClasses[below],
        xSize && xClasses[xSize],
        ySize && yClasses[squish ? exports.squishClass[ySize] : ySize],
        between && inline && inlineContainerClass,
        between && inline && inlineClasses[between],
        between && !inline && stackClasses[between],
    ].filter(Boolean).join(' ');
    const Tag = tag || (inline ? 'span' : 'div');
    return React.createElement(Tag, Object.assign({ className: classes }, tagProps), children);
}
exports.Panel = Panel;
exports.default = Panel;
