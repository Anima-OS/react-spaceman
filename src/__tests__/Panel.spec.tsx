import { shallow } from 'enzyme';
import * as React from 'react';
import { Panel } from '../Panel';

describe('Panel', () => {
	it('should render a div', () => {
		const wrapper = shallow(<Panel>hello</Panel>);
		expect(wrapper).toMatchSnapshot();
	});

	it('should add a class for margin below', () => {
		const wrapper = shallow(<Panel below="m">hello</Panel>);
		expect(wrapper.prop('className')).toBe('rspcmn-b-m');
	});

	it('should add a class for horizontal padding', () => {
		const wrapper = shallow(<Panel x="m">hello</Panel>);
		expect(wrapper.prop('className')).toBe('rspcmn-x-m');
	});

	it('should add a class for vertical padding', () => {
		const wrapper = shallow(<Panel y="m">hello</Panel>);
		expect(wrapper.prop('className')).toBe('rspcmn-y-m');
	});

	it('should add a class for padding', () => {
		const wrapper = shallow(<Panel inset="m">hello</Panel>);
		expect(wrapper.prop('className')).toBe('rspcmn-x-m rspcmn-y-m');
	});

	it('should add a class for squished padding', () => {
		const wrapper = shallow(
			<Panel inset="m" squish>
				hello
			</Panel>
		);
		expect(wrapper.prop('className')).toBe('rspcmn-x-m rspcmn-y-s');
	});

	it('should add a class for margin between children', () => {
		const wrapper = shallow(
			<Panel between="m">
				<div>ein</div>
				<div>zwei</div>
				<div>polizei</div>
			</Panel>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('should add a class for margin between inline children', () => {
		const wrapper = shallow(
			<Panel between="m" inline>
				<span>ein</span>
				<span>zwei</span>
				<span>polizei</span>
			</Panel>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('should add a custom class name', () => {
		const wrapper = shallow(
			<Panel below="m" className="pizza">
				hello
			</Panel>
		);
		expect(wrapper.prop('className')).toBe('pizza rspcmn-b-m');
	});

	it('should render a custom tag', () => {
		const wrapper = shallow(<Panel tag="header">hello</Panel>);
		expect(wrapper).toMatchSnapshot();
	});

	it('should pass props to a custom tag', () => {
		const wrapper = shallow(
			<Panel tag="header" tagProps={{ id: 'pizza' }}>
				hello
			</Panel>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
