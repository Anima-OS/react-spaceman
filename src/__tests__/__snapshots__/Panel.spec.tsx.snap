// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should add a class for margin between children 1`] = `
<div
  className="rspcmn-s-m"
>
  <div>
    ein
  </div>
  <div>
    zwei
  </div>
  <div>
    polizei
  </div>
</div>
`;

exports[`Panel should add a class for margin between inline children 1`] = `
<span
  className="rspcmn-i rspcmn-i-m"
>
  <span>
    ein
  </span>
  <span>
    zwei
  </span>
  <span>
    polizei
  </span>
</span>
`;

exports[`Panel should pass props to a custom tag 1`] = `
<header
  className=""
  id="pizza"
>
  hello
</header>
`;

exports[`Panel should render a custom tag 1`] = `
<header
  className=""
>
  hello
</header>
`;

exports[`Panel should render a div 1`] = `
<div
  className=""
>
  hello
</div>
`;
